src/Psalm/Internal/Analyzer/Statements/Expression/Fetch/ConstFetchAnalyzer.php

```diff
@@ -164,13 +167,22 @@ public static function getGlobalConstType(
                 case 'PHP_MAJOR_VERSION':
                 case 'PHP_MINOR_VERSION':
                 case 'PHP_RELEASE_VERSION':
-                case 'PHP_DEBUG':
                 case 'PHP_FLOAT_DIG':
+                    return Type::getInt();
+
                 case 'PHP_INT_MIN':
+                    return Type::getInt(false, PHP_INT_MIN);
+
+                case 'PHP_DEBUG':
                 case 'PHP_ZTS':
-                    return Type::getInt();
+                    if ($codebase->analysis_php_version_id >= 8_04_00) {
+                        return Type::getBool();
+                    }
+                    return Type::getIntRange(0, 1);
 
                 case 'PHP_INT_MAX':
+                    return Type::getInt(false, PHP_INT_MAX);
```

Comment:
This and PHP_INT_MIN should not be a literal, instead it should be a generic positive integer, because i.e. running psalm on 32-bit and 64-bit systems will return different results.

Reply:
PHP_INT_MAX is already correct at least on master, PHP_INT_MIN needs Type::getIntRange(null, -1)